### PR TITLE
[OYPD-281] Demo camps and their landing pages

### PR DESCRIPTION
**Pull request by hamrant**
Jira - https://propeople-us.atlassian.net/browse/OYPD-281
drupal.org issue https://www.drupal.org/node/2855182

Steps for review:
- [x]  login as admin
- [x]  go to /admin/content?status=All&type=camp&title=&langcode=All
- [x]  check created camps
- [x]  go to /admin/content?status=All&type=landing_page&title=&langcode=All
- [x]  check langing pages for camps
- [x]  profit

### CODE REVIEW

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build565

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/565/

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build565/profiles/openy/build/reports/behat

**Comment by alexschedrov** [comment]
image

**Review by alexschedrov — CHANGES_REQUESTED** [review]
Fixed above.

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build569

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/569/

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build569/profiles/openy/build/reports/behat

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build571

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/571/

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build571/profiles/openy/build/reports/behat

File: modules/openy_features/openy_demo_content/modules/openy_demo_nlanding/config/install/migrate_plus.migration.node_landing.yml [hunk 1]
```diff
@@ -55,10 +60,35 @@ source:
...
       title: 'Locations'
+      alias: '/locations'
       header_content_id: locations_location_finder_filters
       content_ids:
         -
           content_id: locations_location_finder
+    -
+      id: lp_camp_page_1
+      title: 'Camp Colman page'
+      alias: '/lp-camp-page-1'
```
Comment (alexschedrov) [inline modules/openy_features/openy_demo_content/modules/openy_demo_nlanding/config/install/migrate_plus.migration.node_landing.yml]: Let's set better alias and for all pages below.
/[camp-alias]/about

File: modules/openy_features/openy_demo_content/modules/openy_demo_nlanding/config/install/migrate_plus.migration.lp_paragraph_simple.yml [hunk 2]
```diff
@@ -54,6 +54,30 @@ source:
         and more than $9.5 million in financial assistance to more than 27,000
         people to ensure everyone has a chance to succeed and belong at the Y.
         </h4>
+    -
+      id: camp_page_1_simple
+      parent_id: lp_camp_page_1
+      description: |
+        <h4>Do you love Camp Colman? Or maybe you have never been here before
```
Comment (alexschedrov) [inline modules/openy_features/openy_demo_content/modules/openy_demo_nlanding/config/install/migrate_plus.migration.lp_paragraph_simple.yml:61]: Let's add more text here and below. You can take some examples from Seattle.

**Review by alexschedrov — CHANGES_REQUESTED** [review]

**Comment by alexschedrov** [comment]
1. We should support upgrade path.
In profile enable this module and run migration in case if migration hasn't be triggered during installation

**Comment by bjwilkins** [comment]
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build577

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/577/

**Comment by bjwilkins** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build577/profiles/openy/build/reports/behat

File: openy.install [hunk 3]
```diff
@@ -23,3 +23,18 @@ function openy_install() {
   // Enable the admin theme.
   \Drupal::configFactory()->getEditable('node.settings')->set('use_admin_theme', TRUE)->save(TRUE);
 }
+
+/**
+ * Upgrade from 8.x-0.2 version.
+ */
+function openy_update_8001() {
+  // Enable modules.
+  \Drupal::service('module_installer')->install([
```
Comment (alexschedrov) [inline openy.install]: We should do that only in case if `openy_migrate` has been enabled.

File: openy.install [hunk 4]
```diff
@@ -23,3 +23,18 @@ function openy_install() {
...
+    'openy_demo_ncamp',
+  ], TRUE);
+
+  // Run openy_demo_ncamp migration.
+  $importer = \Drupal::service('openy_migrate.importer');
```
Comment (alexschedrov) [inline openy.install]: We should do that only in case if `openy_migrate` has been enabled.

**Review by alexschedrov — CHANGES_REQUESTED** [review]

**Comment by bjwilkins** [comment]
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build578

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/578/

**Comment by bjwilkins** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build578/profiles/openy/build/reports/behat

**Review by alexschedrov — APPROVED** [review]
Approved.

**Pull request merged**